toolchains/kubectl/kubectl_configure.bzl

```diff
@@ -16,16 +16,79 @@ Defines a repository rule for configuring the kubectl tool.
 """
 
 def _impl(repository_ctx):
-
-    kubectl_tool_path = repository_ctx.which("kubectl")
+    substitutions = None
+    label = None
+    if repository_ctx.attr.build_srcs:
+        kubectl_target = "@io_kubernetes//cmd/kubectl:kubectl"
+        substitutions = {"%{KUBECTL_TARGET}": "%s" % kubectl_target}
+        template = Label("@io_bazel_rules_k8s//toolchains/kubectl:BUILD.target.tpl")
+    else:
+        kubectl_tool_path = repository_ctx.which("kubectl")
+        substitutions = {"%{KUBECTL_TOOL}": "%s" % kubectl_tool_path}
+        template = Label("@io_bazel_rules_k8s//toolchains/kubectl:BUILD.path.tpl")
 
     repository_ctx.template(
         "BUILD",
-        Label("@io_bazel_rules_k8s//toolchains/kubectl:BUILD.tpl"),
-        {"%{KUBECTL_TOOL}": "%s" % kubectl_tool_path},
-        False
+        template,
+        substitutions,
+        False,
     )
 
-kubectl_configure = repository_rule(
+_kubectl_configure = repository_rule(
     implementation = _impl,
+    attrs = {
+        "build_srcs": attr.bool(
+            doc = "Optional. Set to true to build kubectl from sources.",
+            default = False,
+            mandatory = False,
+        ),
+    },
 )
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def kubectl_configure(name, **kwargs):
+    """
+    Creates an external repository with a kubectl_toolchain target
+    properly configured.
+
+    Args:
+        **kwargs:
+      Required Args
+        name: A unique name for this rule.
+      Default Args:
+        build_srcs: Optional. Set to true to build kubectl from sources. Default: False
+        k8s_commit: Otional. Commit / release tag at which to build kubectl
+          from. Default "v1.13.0-beta.1"
```

Comment:
we need an issue to update this when 1.13 is out.

Also we need docs that only = or gt 1.13 is supported

Reply:
will do docs in upcoming PR to avoid re-triggering CI.